base/inference.jl

```diff
@@ -2281,6 +2281,17 @@ function inlineable(f::ANY, ft::ANY, e::Expr, atypes::Vector{Any}, sv::Inference
     metharg = meth[1]::Type
     methsp = meth[2]
     method = meth[3]::Method
+    ambig = method.ambig
+    if ambig != nothing
```

Comment:
I think handling this should be the responsibility of `ml_matches`. It can choose either to return no matches if there is definitely an ambiguity, or multiple matches if it isn't sure. Either way inlining will automatically be rejected.